### PR TITLE
Fix up remaining WTR tests

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

* found an edge case with pretty printed tabs in vertical mode
* brought back asm, source, and returnvalue tests
* fixed some smaller timing things
* reformated some files

### Test Plan

the tests ran 50 times each and we had 3 failures in total. that's a pretty low failure rate

### CODE REVIEW

File: src/components/Editor/Tabs.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ function getHiddenTabs(sourceTabs, sourceTabEls) {
 
   const tabTopOffset = getTopOffset();
   return sourceTabs.filter((tab, index) => {
-    return sourceTabEls[index].getBoundingClientRect().top > tabTopOffset;
+    return sourceTabEls[index].getBoundingClientRect().top > tabTopOffset + 10;
```
Comment (jasonLaster) [inline src/components/Editor/Tabs.js]: adjusts for selected sources being offset by the border...

File: src/components/SecondaryPanes/Frames.js [hunk 2]
```diff
@@ -221,14 +221,10 @@ function getAndProcessFrames(state) {
   frames = frames
     .toJS()
     .filter(frame => getSourceForFrame(state, frame))
-    .map(frame =>
-      Object.assign({}, frame, {
-        source: getSourceForFrame(state, frame).toJS()
-      })
-    )
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/Frames.js]: is made redundant by `appendSource`

File: src/test/integration/runner.js [hunk 3]
```diff
@@ -33,133 +33,80 @@ describe("Tests", () => {
...
 
-  xit("asm", async function() {
-    await asm(ctx);
+  afterEach(() => {
+    prefs.pauseOnExceptions = false;
+    prefs.ignoreCaughtExceptions = false;
+    prefs.pendingSelectedLocation = {};
+    prefs.expressions = [];
+    prefs.pendingBreakpoints = [];
+    prefs.tabs = [];
```
Comment (jasonLaster) [inline src/test/integration/runner.js]: helps in some race conditions

File: src/test/integration/runner/styles.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  height: 700px;
+}
+
+iframe {
+  width: 100%;
+  height: 100%;
+  border: 0;
+  background: white;
+}
+}
```
Comment (jasonLaster) [inline src/test/integration/runner/styles.css]: moved files and exctracted css

File: src/utils/frame.js [hunk 5]
```diff
@@ -4,19 +4,20 @@ import { endTruncateStr } from "./utils";
 
 import type { Frame } from "debugger-html";
 
+function getFrameUrl(frame) {
+  return get(frame, "source.url", "") || "";
+}
```
Comment (jasonLaster) [inline src/utils/frame.js]: was returning null for sources that didn't have an URL

File: src/utils/utils.js [hunk 6]
```diff
@@ -74,6 +74,10 @@ function waitForMs(ms: number) {
 
 type duplicatesPredicate = (any, any) => boolean;
 function filterDuplicates(list: Object[], predicate: duplicatesPredicate) {
+  if (list.length == 0) {
+    return [];
+  }
```
Comment (jasonLaster) [inline src/utils/utils.js]: was returning `[undefined]` for empty lists previously
